modules/javafx.graphics/src/main/java/javafx/print/JobSettings.java

```diff
@@ -462,6 +462,97 @@ public void setJobName(String name) {
     }
     ///////////////////////  END JOBNAME /////////////////////
 
+    ///////////////////////  START OUTPUTFILE /////////////////////
+
+    private static final String DEFAULT_OUTPUTFILE = "";
+    private SimpleStringProperty outputFile;
+
+    /**
+     * A {@code StringProperty} representing the {@code java.net.URL}
+     * encoded name of a filesystem file, to which the platform printer
+     * driver should spool the rendered print data.
+     * <p>
+     * Applications can use this to request print-to-file behaviour where
+     * the native print system is capabale of spooling the output to a
+     * filesystem file, rather than the printer device.
+     * <p>
+     * This is often useful where the printer driver generates a format
+     * such as Postscript or PDF, and the application intends to distribute
+     * the result instead of printing it, or for some other reason the
+     * application does not want physical media (paper) emitted by the printer.
+     * <p>
+     * If the application does not have permission to write to the specified
+     * file, a {@code SecurityException} may be thrown when printing.
+     * If the print system does not support print-to-file, then this
+     * <p>
+     * setting will be ignored.
+     * If the URL specifies a non-existent path, or does not specify
+     * a writable file it may be ignored, or a SecurityException may be thrown.
```

Comment:
I want to look at what we do for other properties like this. I'm not sure how feasible it is to throw an exception back to the application. And I wouldn't want to single out a security exception (what about a read-only file system?). I think that whatever the reason, if the file can't be written to, we should treat it consistently. Probably by ignoring the request. I presume that's what will happen if the print driver doesn't support print-to-file?

Reply:
Not easy to specify exactly. We are handing it off to 2D
We aren't doing any up-front verification here, but I don't want to copy 2D behaviour into the docs, even if I could be sure about that .. I may just want to be more waffly about what happens.
Could be printed to the printer, could be a failed job, could be an exception of some kind if some nice piece of Java code recognises a problem up-front